input/fsh/UpdateBundle.fsh

```diff
@@ -7,7 +7,10 @@ Description: """
     communicated in FHIRcast `-update` messages. The bundle can only contain
     requests of type PUT and DELETE.  
     POST is not allowed as the content sharing mechanism cannot indicate the 
-    id of the created resource.
+    id of a resource created by a POST method.  Since the creator of the resource
+    does wish to have control over the id of a resource it has created, as per the FHIR
```

Comment:
I would leave this text in place.

Reply:
The root cause behind this whole discussion is that the source Subscriber sends out a requested change all the Subscribers have to respond to. The result is that the new id of the resources that are created are not communicated. An alternative approach would be to treat the request to update and the event that is send out differently.

The request to update would hold a batch (or transaction) bundle with the requested updates. Upon receiving the request, the Hub would process them in the normal FHIR way (see https://build.fhir.org/http.html#transaction). Instead of forwarding the request, the Hub should forward the result of the request as a history Bundle. This bundle shows the request and outcome of each requested change. 

This approach makes the content exchange behavior very close to the behavior of a normal FHIR server. All subscribers will receive the request as well as the result of the request. So they know whether an operation succeeded or not. They will also know what id was assigned to a newly created resource.

Reply:
If the "beginning" thought would that:

1. the Subscriber providing the resource sends a resource in an entry of a transaction Bundle with the entry having a method of POST
2. the resource has no id, the Hub assigns and id then distributes the event with populated id's

I ran that by Nuance and our internal group.  They were not enthusiastic since it puts more responsibility on the Hub.  Hence the thought of adding a sentence indicating that the provider of the resource populate the id with something it believes to be globally unique therefore usable by everyone.  It does not address the "what if the resource is persisted to a FHIR server and the FHIR server replaces the id with its own id" scenario; however, neither would the Hub assigning a resource id.

Regarding the history Bundle, I read through that section.  It is an interesting "extra" but I don't believe our implementors would be very enthusiastic about it.  At best I believe they would ignore the response, so I am not sure that gains us anything.